src/Raven.Server/Commercial/LicenseManager.cs

```diff
@@ -262,6 +259,21 @@ public void ReloadLicenseLimits(bool addPerformanceHint = false)
             }
         }
 
+        public bool GetCoresLimitForNode(out int utilizedCores)
+        {
+            var licenseLimits = _serverStore.LoadLicenseLimits();
+            if (licenseLimits?.NodeLicenseDetails != null &&
+                licenseLimits.NodeLicenseDetails.TryGetValue(_serverStore.NodeTag, out var detailsPerNode))
+            {
+                utilizedCores = Math.Min(detailsPerNode.UtilizedCores, _licenseStatus.MaxCores);
+                return true;
+                
+            }
+
+            utilizedCores = _licenseStatus.MaxCores;
```

Comment:
Shouldn't we use ProcessorCount instead of MaxCores?

Reply:
No. We get here if:
- First start of the server.
- We didn't apply the license limits yet.

Reply:
But you cannot have more utilized cores than ProcessorCount. Shouldn't this be `Math.Min(Environment.ProcessorCount, _licenseStatus.MaxCores)` ?